pkg/launch/launchable.go

```diff
@@ -49,6 +49,16 @@ type LaunchableStanza struct {
 	// "always".
 	RestartPolicy_ runit.RestartPolicy `yaml:"restart_policy,omitempty"`
 
+	// NoHaltOnUpdate instructs the preparer to skip stopping the
+	// launchable's processes when it is being updated. This is useful for
+	// processes that are designed to be updated via binary overwrite and
+	// SIGHUP for example.
+	// NOTE: under certain circumstances the process still might be
+	// stopped, for example if a host operator calls p2-shutdown
+	// NOTE: P2 does not arrange for signals to your app to signify update,
+	// that should be implemented by the pod itself via bin/post-activate
+	NoHaltOnUpdate bool `yaml:"no_halt_on_update,omitempty"`
```

Comment:
i'm happy to discuss the naming here. Notably this setting shouldn't be ported to container launchables because containers should be immutable